dev/release/verify-release-candidate.sh

```diff
@@ -880,8 +880,8 @@ test_csharp() {
   if [ "${SOURCE_KIND}" = "local" ]; then
     echo "Skipping sourelink verification on local build"
   else
-    dotnet tool run sourcelink test artifacts/Apache.Arrow/Release/netstandard1.3/Apache.Arrow.pdb
-    dotnet tool run sourcelink test artifacts/Apache.Arrow/Release/netcoreapp3.1/Apache.Arrow.pdb
+    dotnet tool run sourcelink test artifacts/Apache.Arrow/Release/netstandard2.0/Apache.Arrow.pdb
```

Comment:
I have no experience with C# and I might be saying something that doesn't make sense but is there a reason why we don't test this when `SOURCE_KIND` is `local`?
We probably could find the issue on nightlies / maintenance branch if we also tested there instead of only when we have generated a Release candidate.